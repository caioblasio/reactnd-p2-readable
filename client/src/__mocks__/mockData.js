const testPosts = {

  "8xf0y6ziyjabvozdd253nd": {
    id: '8xf0y6ziyjabvozdd253nd',
    timestamp: 1467166872634,
    title: 'Udacity is the best place to learn React',
    body: 'Everyone says so after all.',
    author: 'thingtwo',
    category: 'categoryone',
    voteScore: 6,
    deleted: false,
    commentCount: 2
  },
  "6ni6ok3ym7mf1p33lnez": {
    id: '6ni6ok3ym7mf1p33lnez',
    timestamp: 1468479767190,
    title: 'Learn Redux in 10 minutes!',
    body: 'Just kidding. It takes more than 10 minutes to learn technology.',
    author: 'thingone',
    category: 'categorytwo',
    voteScore: -5,
    deleted: false,
    commentCount: 0
  }
}

const testCategories = [
  {
    name: 'Category One',
    path: 'categoryone'
  },
  {
    name: 'Category Two',
    path: 'categorytwo'
  }
]

const testComments = [
  {
    id: '894tuq4ut84ut8v4t8wun89g',
    parentId: "8xf0y6ziyjabvozdd253nd",
    timestamp: 1468166872634,
    body: 'Hi there! I am a COMMENT.',
    author: 'thingtwo',
    voteScore: 6,
    deleted: false,
    parentDeleted: false
  },
  {
    id: '8tu4bsun805n8un48ve89',
    parentId: "8xf0y6ziyjabvozdd253nd",
    timestamp: 1469479767190,
    body: 'Comments. Are. Cool.',
    author: 'thingone',
    voteScore: -5,
    deleted: false,
    parentDeleted: false
  }
  
]

/* JSON headers for API requests */
const jsonHeaders = new Headers({
	Accept: 'application/json',
	'Content-Type': 'application/json'
});

export { testPosts, testCategories, testComments, jsonHeaders };