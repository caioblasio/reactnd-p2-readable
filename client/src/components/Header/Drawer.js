import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import Drawer from '@material-ui/core/Drawer';
import List from '@material-ui/core/List';
import Hidden from '@material-ui/core/Hidden';
import Divider from '@material-ui/core/Divider';


import ListSubheader from '@material-ui/core/ListSubheader';
import ListItem from '@material-ui/core/ListItem';
import ListItemIcon from '@material-ui/core/ListItemIcon';
import ListItemText from '@material-ui/core/ListItemText';
import NoteAdd from '@material-ui/icons/NoteAdd';
import StarIcon from '@material-ui/icons/Star';


import { Link } from 'react-router-dom';

const styles = theme => ({
  navIconHide: {
    [theme.breakpoints.up('md')]: {
      display: 'none',
    },
  },
  drawerPaper: {
    width: theme.drawerWidth,
    flexShrink: 0,
  },
  content: {
    flexGrow: 1,
    backgroundColor: theme.palette.background.default,
    padding: theme.spacing.unit * 3,
  },
  link: {
    ...theme.link
  },
  toolbar: theme.mixins.toolbar,
});

class ResponsiveDrawer extends React.Component {

  render() {
    const { isOpen, categories, toggleDrawer, classes, theme } = this.props;

    const drawer = (
      <Fragment>
        <div className={classes.toolbar} />
        <Divider />
        <List
          component="nav"
          subheader={<ListSubheader component="div">Categories</ListSubheader>}
        >
          <Link to="/" className={classes.link}>
            <ListItem button>
              <ListItemText primary="All" />
            </ListItem>
          </Link>
          {categories.map(category => (
            <Link key={category.path} to={`/${category.path}`} className={classes.link}>
              <ListItem key={category.path} button>
                <ListItemText primary={category.name} />
              </ListItem>
            </Link>
          ))}
        </List>
        <Divider />
        <List
          component="nav"
          subheader={<ListSubheader component="div">Posts</ListSubheader>}
        >
          <Link to="/new" className={classes.link}>
            <ListItem button>
              <ListItemIcon>
                <NoteAdd />
              </ListItemIcon>
              <ListItemText primary="Create Post" />
            </ListItem>
          </Link>
          <ListItem button>
            <ListItemIcon>
              <StarIcon />
            </ListItemIcon>
            <ListItemText primary="Starred" />
          </ListItem>
        </List>
      </Fragment>
    );

    return (
      <Fragment>
        <Hidden mdUp>
          <Drawer
            variant="temporary"
            anchor={theme.direction === 'rtl' ? 'right' : 'left'}
            open={isOpen}
            onClose={toggleDrawer}
            classes={{
              paper: classes.drawerPaper,
            }}
            ModalProps={{
              keepMounted: true, // Better open performance on mobile.
            }}
          >
            {drawer}
          </Drawer>
        </Hidden>
        <Hidden smDown implementation="css">
          <Drawer
            variant="persistent"
            open={isOpen}
            classes={{
              paper: classes.drawerPaper,
            }}
          >
            {drawer}
          </Drawer>
        </Hidden>
      </Fragment>
    );
  }
}

ResponsiveDrawer.propTypes = {
  classes: PropTypes.object.isRequired,
  theme: PropTypes.object.isRequired,
  isOpen: PropTypes.bool,
};

export default withStyles(styles, { withTheme: true })(ResponsiveDrawer);