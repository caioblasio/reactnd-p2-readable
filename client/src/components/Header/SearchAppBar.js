import React, { Component, Fragment } from 'react';
import PropTypes from 'prop-types';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import IconButton from '@material-ui/core/IconButton';
import Typography from '@material-ui/core/Typography';
import InputBase from '@material-ui/core/InputBase';
import { fade } from '@material-ui/core/styles/colorManipulator';
import { withStyles } from '@material-ui/core/styles';
import MenuIcon from '@material-ui/icons/Menu';
import SearchIcon from '@material-ui/icons/Search';

import { Link } from 'react-router-dom';
import { connect } from 'react-redux';

const styles = theme => ({

  appBar: {
    zIndex: theme.zIndex.drawer + 1,
  },
  grow: {
    flexGrow: 1,
  },
  menuButton: {
    marginLeft: -12,
    marginRight: 20,
  },
  title: {
    display: 'none',
    [theme.breakpoints.up('sm')]: {
      display: 'block',
    },
  },
  link: {
    ...theme.link,
  },
  search: {
    position: 'relative',
    borderRadius: theme.shape.borderRadius,
    backgroundColor: fade(theme.palette.common.white, 0.15),
    '&:hover': {
      backgroundColor: fade(theme.palette.common.white, 0.25),
    },
    marginLeft: 0,
    width: '100%',
    [theme.breakpoints.up('sm')]: {
      marginLeft: theme.spacing.unit,
      width: 'auto',
    },
  },
  searchIcon: {
    width: theme.spacing.unit * 9,
    height: '100%',
    position: 'absolute',
    pointerEvents: 'none',
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
  },
  inputRoot: {
    color: 'inherit',
    width: '100%',
  },
  inputInput: {
    paddingTop: theme.spacing.unit,
    paddingRight: theme.spacing.unit,
    paddingBottom: theme.spacing.unit,
    paddingLeft: theme.spacing.unit * 10,
    transition: theme.transitions.create('width'),
    width: '100%',
    [theme.breakpoints.up('sm')]: {
      width: 180,
      '&:focus': {
        width: 240,
      },
    },
  },
});



class SearchAppBar extends Component {

  state = {
    query: '',
    typingTimeout: 0
  };

  /**
  * @description Handles change at input field with debounce
  * @param {object} event
  */
  handleChange = (event) => {

    const { changeSearch } = this.props;
    const self = this;
    const query = event.target.value;

    if (self.state.typingTimeout) {
      clearTimeout(self.state.typingTimeout);
    }

    this.setState({
      query,
      typingTimeout: setTimeout(function () {
        changeSearch(query.toLowerCase());
      }, 500)
    });
  };

  render() {
    const { isOpen, openDrawer, closeDrawer, classes } = this.props;
    return (
      <Fragment>
        <AppBar position="fixed" className={classes.appBar}>
          <Toolbar>
            <IconButton 
              className={classes.menuButton} 
              color="inherit" 
              aria-label="Open drawer"
              onClick={isOpen ? closeDrawer : openDrawer}
              >
              <MenuIcon />
            </IconButton>
            <Link to="/" className={classes.link}>
              <Typography className={classes.title} variant="h6" color="inherit" noWrap>
                Readable
              </Typography>
            </Link>
            <div className={classes.grow} />
            <div className={classes.search}>
              <div className={classes.searchIcon}>
                <SearchIcon />
              </div>
              <InputBase
                placeholder="Search for postsâ€¦"
                classes={{
                  root: classes.inputRoot,
                  input: classes.inputInput,
                }}
                value={this.state.query}
                onChange={this.handleChange}
              />
            </div>
          </Toolbar>
        </AppBar>
      </Fragment>
    );
  }
}

SearchAppBar.propTypes = {
  classes: PropTypes.object.isRequired,
};

export default withStyles(styles)(connect()(SearchAppBar));