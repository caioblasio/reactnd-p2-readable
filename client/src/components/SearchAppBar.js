import React, { Component, Fragment } from 'react';
import PropTypes from 'prop-types';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import IconButton from '@material-ui/core/IconButton';
import Typography from '@material-ui/core/Typography';
import InputBase from '@material-ui/core/InputBase';
import { fade } from '@material-ui/core/styles/colorManipulator';
import { withStyles } from '@material-ui/core/styles';
import MenuIcon from '@material-ui/icons/Menu';
import SearchIcon from '@material-ui/icons/Search';

import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import { toggleDrawer } from '../actions/drawer';

const styles = theme => ({

  appBar: {
    zIndex: theme.zIndex.drawer + 1,
  },
  grow: {
    flexGrow: 1,
  },
  menuButton: {
    marginLeft: -12,
    marginRight: 20,
  },
  title: {
    display: 'none',
    [theme.breakpoints.up('sm')]: {
      display: 'block',
    },
  },
  link: {
    ...theme.link,
  },
  search: {
    position: 'relative',
    borderRadius: theme.shape.borderRadius,
    backgroundColor: fade(theme.palette.common.white, 0.15),
    '&:hover': {
      backgroundColor: fade(theme.palette.common.white, 0.25),
    },
    marginLeft: 0,
    width: '100%',
    [theme.breakpoints.up('sm')]: {
      marginLeft: theme.spacing.unit,
      width: 'auto',
    },
  },
  searchIcon: {
    width: theme.spacing.unit * 9,
    height: '100%',
    position: 'absolute',
    pointerEvents: 'none',
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
  },
  inputRoot: {
    color: 'inherit',
    width: '100%',
  },
  inputInput: {
    paddingTop: theme.spacing.unit,
    paddingRight: theme.spacing.unit,
    paddingBottom: theme.spacing.unit,
    paddingLeft: theme.spacing.unit * 10,
    transition: theme.transitions.create('width'),
    width: '100%',
    [theme.breakpoints.up('sm')]: {
      width: 120,
      '&:focus': {
        width: 200,
      },
    },
  },
});



class SearchAppBar extends Component {

  handleDrawerToggle = () => {
    this.props.dispatch(toggleDrawer())
  };

  render() {
    const { classes } = this.props;
    return (
      <Fragment>
        <AppBar position="fixed" className={classes.appBar}>
          <Toolbar>
            <IconButton 
              className={classes.menuButton} 
              color="inherit" 
              aria-label="Open drawer"
              onClick={this.handleDrawerToggle}
              >
              <MenuIcon />
            </IconButton>
            <Link to="/" className={classes.link}>
              <Typography className={classes.title} variant="h6" color="inherit" noWrap>
                Readable
              </Typography>
            </Link>
            <div className={classes.grow} />
            <div className={classes.search}>
              <div className={classes.searchIcon}>
                <SearchIcon />
              </div>
              <InputBase
                placeholder="Searchâ€¦"
                classes={{
                  root: classes.inputRoot,
                  input: classes.inputInput,
                }}
              />
            </div>
          </Toolbar>
        </AppBar>
      </Fragment>
    );
  }
}

SearchAppBar.propTypes = {
  classes: PropTypes.object.isRequired,
};

export default withStyles(styles)(connect()(SearchAppBar));