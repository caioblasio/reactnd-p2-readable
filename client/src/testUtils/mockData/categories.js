export const testCategories = [
  {
    name: 'Category One',
    path: 'categoryone'
  },
  {
    name: 'Category Two',
    path: 'categorytwo'
  }
]