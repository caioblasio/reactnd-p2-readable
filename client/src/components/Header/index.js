import React, { Component, Fragment } from 'react';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';
import PropTypes from 'prop-types';
import { openDrawer, closeDrawer } from '../../actions/drawer';
import { changeSearch } from '../../actions/search';
import { fetchCategories } from '../../actions/categories';
import SearchAppBar from './SearchAppBar';
import ResponsiveDrawer from './Drawer';
import { isMobile } from 'react-device-detect';

/**
 * @description Component for adding a component in a post detail page
 * @param {bool} isOpen
 * @param {array} categories
 * @param {function()} openDrawer
 * @param {function()} closeDrawer
 * @param {function()} changeSearch
 * @param {function()} getCategories
 * @param {object} history
*/
class Header extends Component {

  static propType = {
    isOpen: PropTypes.bool.isRequired,
    categories: PropTypes.array.isRequired,
    openDrawer: PropTypes.func.isRequired,
    closeDrawer: PropTypes.func.isRequired,
    changeSearch: PropTypes.func.isRequired,
    getCategories: PropTypes.func.isRequired,
    history: PropTypes.object.isRequired
  };

  /**
   * @description Get categories to display
  */
  componentDidMount(){
    this.props.getCategories();
  }

  /**
   * @description Closes drawer when there is a page change in mobile
   * @param {object} history
   * @param {bool} isOpen
   * @param {function()} closeDrawer
  */
  componentWillReceiveProps({ history, isOpen, closeDrawer }) {   
    history.listen(() => {
      if(isMobile && isOpen)
        closeDrawer();
    });
  }

  render(){
    const { isOpen, categories, openDrawer, closeDrawer, changeSearch } = this.props;
    return (
      <Fragment>
        <SearchAppBar
          changeSearch={changeSearch}
          openDrawer={openDrawer}
          closeDrawer={closeDrawer}
          isOpen={isOpen}
        />
        <ResponsiveDrawer 
          isOpen={isOpen}
          categories={categories}
          openDrawer={openDrawer}
          closeDrawer={closeDrawer}
        />
      </Fragment>
    )
  }
}

const mapStateToProps = ({ drawer, categories }) => ({ isOpen: drawer, categories })
  
const mapDispatchToProps = dispatch => {
  return {
    openDrawer: () => {
      dispatch(openDrawer())
    },
    closeDrawer: () => {
      dispatch(closeDrawer())
    },
    changeSearch: (query) => {
      dispatch(changeSearch(query))
    },
    getCategories: () => {
      dispatch(fetchCategories())
    }
  }
}

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(Header));