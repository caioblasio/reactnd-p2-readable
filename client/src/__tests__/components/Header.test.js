import React from 'react';
import Header from '../../components/Header';
import SearchAppBar from '../../components/Header/SearchAppBar';
import ResponsiveDrawer from '../../components/Header/Drawer';
import toJson from 'enzyme-to-json';
import configureStore from 'redux-mock-store';
import { testPosts, testCategories, jsonHeaders } from '../../__mocks__/mockData';
import thunk from 'redux-thunk'
import { createShallow } from '@material-ui/core/test-utils';
import MemoryRouter from 'react-router-dom/MemoryRouter';
import Route from 'react-router-dom/Route';
import { InputBase } from '@material-ui/core';

const middlewares = [ thunk ];
const mockStore = configureStore(middlewares);

describe('<Header />', () => {

  let wrapper, store;
  
  beforeAll(() => {
    const initialState = {
      categories: testCategories,
      drawer: false,
    }
    store = mockStore(initialState);
  });

  beforeEach(() => {
    fetch.resetMocks();
  });

  it('renders correctly', () => {

    let shallow = createShallow({untilSelector: Route});

    fetch
      .mockResponse(JSON.stringify(testCategories), { jsonHeaders })

    wrapper = shallow(
      <MemoryRouter>
        <Header
          store={store}
        />
      </MemoryRouter>
    );

    const component = wrapper.dive().dive()
    expect(toJson(component)).toMatchSnapshot();

  });

});

describe('<ResponsiveDrawer />', () => {

  let wrapper;

  it('renders correcty', () => {

    let shallow = createShallow({untilSelector: ResponsiveDrawer})
    const openDrawer = jest.fn(),
      closeDrawer = jest.fn();

    wrapper = shallow(
      <MemoryRouter>
        <ResponsiveDrawer
          isOpen={true}
          categories={testCategories}
          openDrawer={openDrawer}
          closeDrawer={closeDrawer}
        />
      </MemoryRouter>
    );

    const component = wrapper.dive();
    expect(toJson(component)).toMatchSnapshot();

  });

});

describe('<SearchAppBar />', () => {

  let wrapper;

  it('renders correcty', () => {

    let shallow = createShallow({untilSelector: SearchAppBar})
    const openDrawer = jest.fn(),
      closeDrawer = jest.fn(),
      changeSearch = jest.fn();

    wrapper = shallow(
      <MemoryRouter>
        <SearchAppBar
          isOpen={true}
          changeSearch={changeSearch}
          openDrawer={openDrawer}
          closeDrawer={closeDrawer}
        />
      </MemoryRouter>
    );

    const component = wrapper.dive();
    expect(toJson(component)).toMatchSnapshot();

  });

  it('calls change search', (done) => {

    let shallow = createShallow({untilSelector: SearchAppBar})
    const openDrawer = jest.fn(),
      closeDrawer = jest.fn(),
      changeSearch = jest.fn(),
      query = 'search for posts';
    

    wrapper = shallow(
      <MemoryRouter>
        <SearchAppBar
          isOpen={true}
          changeSearch={changeSearch}
          openDrawer={openDrawer}
          closeDrawer={closeDrawer}
        />
      </MemoryRouter>
    );

    const component = wrapper.dive();
    component.find(InputBase).simulate('change', { target: { value: query } });
    expect(component.state().query).toEqual(query)

    setTimeout(() => {
      expect(changeSearch.mock.calls[0][0]).toEqual(query)
      done();
    }, 500);

  });

});