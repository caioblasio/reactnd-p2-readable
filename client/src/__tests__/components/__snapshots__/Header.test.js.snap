// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> renders correctly 1`] = `
<Fragment>
  <WithStyles(SearchAppBar)
    changeSearch={[Function]}
    closeDrawer={[Function]}
    isOpen={false}
    openDrawer={[Function]}
  />
  <WithStyles(ResponsiveDrawer)
    categories={
      Array [
        Object {
          "name": "Category One",
          "path": "categoryone",
        },
        Object {
          "name": "Category Two",
          "path": "categorytwo",
        },
      ]
    }
    closeDrawer={[Function]}
    isOpen={false}
    openDrawer={[Function]}
  />
</Fragment>
`;

exports[`<ResponsiveDrawer /> renders correcty 1`] = `
<Fragment>
  <Hidden
    implementation="js"
    lgDown={false}
    lgUp={false}
    mdDown={false}
    mdUp={true}
    smDown={false}
    smUp={false}
    xlDown={false}
    xlUp={false}
    xsDown={false}
    xsUp={false}
  >
    <WithStyles(Drawer)
      ModalProps={
        Object {
          "keepMounted": true,
        }
      }
      anchor="left"
      classes={
        Object {
          "paper": "ResponsiveDrawer-drawerPaper-2",
        }
      }
      onClose={[MockFunction]}
      open={true}
      variant="temporary"
    >
      <div
        className="ResponsiveDrawer-toolbar-5"
      />
      <WithStyles(Divider) />
      <WithStyles(List)
        component="nav"
        subheader={
          <WithStyles(ListSubheader)
            component="div"
          >
            Categories
          </WithStyles(ListSubheader)>
        }
      >
        <Link
          className="ResponsiveDrawer-link-4"
          replace={false}
          to="/"
        >
          <WithStyles(ListItem)
            button={true}
          >
            <WithStyles(ListItemText)
              primary="All"
            />
          </WithStyles(ListItem)>
        </Link>
        <Link
          className="ResponsiveDrawer-link-4"
          key="categoryone"
          replace={false}
          to="/categoryone"
        >
          <WithStyles(ListItem)
            button={true}
            key="categoryone"
          >
            <WithStyles(ListItemText)
              primary="Category One"
            />
          </WithStyles(ListItem)>
        </Link>
        <Link
          className="ResponsiveDrawer-link-4"
          key="categorytwo"
          replace={false}
          to="/categorytwo"
        >
          <WithStyles(ListItem)
            button={true}
            key="categorytwo"
          >
            <WithStyles(ListItemText)
              primary="Category Two"
            />
          </WithStyles(ListItem)>
        </Link>
      </WithStyles(List)>
      <WithStyles(Divider) />
      <WithStyles(List)
        component="nav"
        subheader={
          <WithStyles(ListSubheader)
            component="div"
          >
            Posts
          </WithStyles(ListSubheader)>
        }
      >
        <Link
          className="ResponsiveDrawer-link-4"
          replace={false}
          to="/new"
        >
          <WithStyles(ListItem)
            button={true}
          >
            <WithStyles(ListItemIcon)>
              <pure(NoteAddIcon) />
            </WithStyles(ListItemIcon)>
            <WithStyles(ListItemText)
              primary="Create Post"
            />
          </WithStyles(ListItem)>
        </Link>
        <Link
          className="ResponsiveDrawer-link-4"
          replace={false}
          to="/favorites"
        >
          <WithStyles(ListItem)
            button={true}
          >
            <WithStyles(ListItemIcon)>
              <pure(StarIcon) />
            </WithStyles(ListItemIcon)>
            <WithStyles(ListItemText)
              primary="Favorites"
            />
          </WithStyles(ListItem)>
        </Link>
      </WithStyles(List)>
    </WithStyles(Drawer)>
  </Hidden>
  <Hidden
    implementation="css"
    lgDown={false}
    lgUp={false}
    mdDown={false}
    mdUp={false}
    smDown={true}
    smUp={false}
    xlDown={false}
    xlUp={false}
    xsDown={false}
    xsUp={false}
  >
    <WithStyles(Drawer)
      classes={
        Object {
          "paper": "ResponsiveDrawer-drawerPaper-2",
        }
      }
      open={true}
      variant="persistent"
    >
      <div
        className="ResponsiveDrawer-toolbar-5"
      />
      <WithStyles(Divider) />
      <WithStyles(List)
        component="nav"
        subheader={
          <WithStyles(ListSubheader)
            component="div"
          >
            Categories
          </WithStyles(ListSubheader)>
        }
      >
        <Link
          className="ResponsiveDrawer-link-4"
          replace={false}
          to="/"
        >
          <WithStyles(ListItem)
            button={true}
          >
            <WithStyles(ListItemText)
              primary="All"
            />
          </WithStyles(ListItem)>
        </Link>
        <Link
          className="ResponsiveDrawer-link-4"
          key="categoryone"
          replace={false}
          to="/categoryone"
        >
          <WithStyles(ListItem)
            button={true}
            key="categoryone"
          >
            <WithStyles(ListItemText)
              primary="Category One"
            />
          </WithStyles(ListItem)>
        </Link>
        <Link
          className="ResponsiveDrawer-link-4"
          key="categorytwo"
          replace={false}
          to="/categorytwo"
        >
          <WithStyles(ListItem)
            button={true}
            key="categorytwo"
          >
            <WithStyles(ListItemText)
              primary="Category Two"
            />
          </WithStyles(ListItem)>
        </Link>
      </WithStyles(List)>
      <WithStyles(Divider) />
      <WithStyles(List)
        component="nav"
        subheader={
          <WithStyles(ListSubheader)
            component="div"
          >
            Posts
          </WithStyles(ListSubheader)>
        }
      >
        <Link
          className="ResponsiveDrawer-link-4"
          replace={false}
          to="/new"
        >
          <WithStyles(ListItem)
            button={true}
          >
            <WithStyles(ListItemIcon)>
              <pure(NoteAddIcon) />
            </WithStyles(ListItemIcon)>
            <WithStyles(ListItemText)
              primary="Create Post"
            />
          </WithStyles(ListItem)>
        </Link>
        <Link
          className="ResponsiveDrawer-link-4"
          replace={false}
          to="/favorites"
        >
          <WithStyles(ListItem)
            button={true}
          >
            <WithStyles(ListItemIcon)>
              <pure(StarIcon) />
            </WithStyles(ListItemIcon)>
            <WithStyles(ListItemText)
              primary="Favorites"
            />
          </WithStyles(ListItem)>
        </Link>
      </WithStyles(List)>
    </WithStyles(Drawer)>
  </Hidden>
</Fragment>
`;

exports[`<SearchAppBar /> renders correcty 1`] = `
<Fragment>
  <WithStyles(AppBar)
    className="SearchAppBar-appBar-6"
    position="fixed"
  >
    <WithStyles(Toolbar)>
      <WithStyles(IconButton)
        aria-label="Open drawer"
        className="SearchAppBar-menuButton-8"
        color="inherit"
        onClick={[MockFunction]}
      >
        <pure(MenuIcon) />
      </WithStyles(IconButton)>
      <Link
        className="SearchAppBar-link-10"
        replace={false}
        to="/"
      >
        <WithStyles(Typography)
          className="SearchAppBar-title-9"
          color="inherit"
          noWrap={true}
          variant="h6"
        >
          Readable
        </WithStyles(Typography)>
      </Link>
      <div
        className="SearchAppBar-grow-7"
      />
      <div
        className="SearchAppBar-search-11"
      >
        <div
          className="SearchAppBar-searchIcon-12"
        >
          <pure(SearchIcon) />
        </div>
        <WithStyles(InputBase)
          classes={
            Object {
              "input": "SearchAppBar-inputInput-14",
              "root": "SearchAppBar-inputRoot-13",
            }
          }
          fullWidth={false}
          inputComponent="input"
          multiline={false}
          onChange={[Function]}
          placeholder="Search for postsâ€¦"
          type="text"
          value=""
        />
      </div>
    </WithStyles(Toolbar)>
  </WithStyles(AppBar)>
</Fragment>
`;
